### ISSUE TITLE
Blacklist entire url tree?

### ISSUE THREAD

**Issue by quietlychris** [madeindjs/spider#32]
Hello,

I was using this library today, and noticed that I was having a lot of issues with one site that had comments enabled. As a result, while the fetch/parsing of the main content was fairly quick, there were many cases where I was also crawling the `https://example.com/article/comment/<comment_number>` as well. When I added in 
```rust
website.configuration.blacklist_url.push("https://www.example.com/comment/".to_string());
```
to my configuration, I believe it successfully stopped that particular page from coming up, but pages that contains that url (although didn't match it exactly) still seemed to appear. It would be nice to be able to avoid this tree entirely (I'm not interested in the comments of the site at all).  

I believe this is because the `is_allowed()` function uses the following internally
```rust
if self.configuration.blacklist_url.contains(link) {
            return false;
        }
``` 
which checks for exact matches of the intended link in the blacklisted urls, rather than checks for a pattern match of the blacklisted url on the link itself. Is there a way that might make sense for helping to prevent this? 

I'm using `spider = "1.5.1"` in my `Cargo.toml` file, with `rustc 1.60.0-nightly`. I've actually seen this behavior on a couple of sites, but at the risk of annoying him by running too many crawlers over it, one example of this might be [`https://www.jeffgeerling.com`](https://www.jeffgeerling.com)

**Comment by j-mendez** [comment]
Hi @quietlychris does this issue happen in 1.5.0?

**Comment by j-mendez** [comment]
@quietlychris pr up with the changes here https://github.com/madeindjs/spider/pull/36. I need to add some test before it gets merged.

**Comment by j-mendez** [comment]
Released in v1.5.3 as an optional feature `regex`. Thanks @quietlychris  🙌
